app/controllers/users_controller.rb

```diff
@@ -11,6 +11,13 @@ def show
     @predictions = @user.predictions
     @statistics  = @user.statistics
     @predictions = @predictions.visible_to_everyone unless user_is_current_user?
+
+    case params[:filter]
+    when 'judged' then @predictions = @predictions.judged
+    when 'unjudged' then @predictions = @predictions.unjudged
+    when 'future' then @predictions = @predictions.future
+    end
```

Comment:
The [airbnb style guide](https://github.com/airbnb/ruby#indent-when-as-case) recommends something like this:
```ruby
@predictions = case params[:filter]
               when 'judged' then @predictions = @predictions.judged
               when 'unjudged' then @predictions = @predictions.unjudged
               when 'future' then @predictions = @predictions.future
               end
```
On the other hand, the possible filters are just names of scopes, so this would also work (with some protection against SQL injection):
```ruby
def show
...
  @predictions = @predictions.send(safe_filter) if safe_filter.present?
...
end

private

def safe_filter 
  @safe_filter ||= [:judged, :unjudged, :future].delete(params[:filter].intern)
end
```

I'm curious what others think. Personally, I'm not a huge fan or re-assigning `@predictions` so many times. Perhaps it's time to pull some of this filtering logic into the model or some kind of decorator.

Reply:
Done! I agree it didn't look too good in the controller, but I also don't want to grow the Prediction model, so I put this in a PredictionFilter module.

Reply:
Nice.